src/ray/core_worker/core_worker.cc

```diff
@@ -419,7 +420,8 @@ Status CoreWorker::Put(const RayObject &object,
       RAY_RETURN_NOT_OK(plasma_store_provider_->Release(object_id));
     }
   }
-  return memory_store_->Put(RayObject(rpc::ErrorType::OBJECT_IN_PLASMA), object_id);
+  RAY_CHECK(memory_store_->Put(RayObject(rpc::ErrorType::OBJECT_IN_PLASMA), object_id));
```

Comment:
Should we just change this to a void return if it can't fial?

Reply:
It's not very nice, but I'm currently using the return value when a task completes to check whether a direct return value was actually promoted to plasma.

Reply:
I see, that wasn't clear from the reading (I thought it was checking a status code). How about passing in a pointer output to set if it was promoted instead?